From c825e273a4b5901e12aaf761849e648a4a2030b3 Mon Sep 17 00:00:00 2001
From: ReinvdZee <rvanderzee@victronenergy.com>
Date: Thu, 11 Jul 2024 13:29:24 +0200
Subject: [PATCH] Fix potential race condition

By explicitly registering on dbus after adding all paths
victronenergy/venus-private#410
---
 startstop.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/startstop.py b/startstop.py
index 1a78072..a33744a 100644
--- a/startstop.py
+++ b/startstop.py
@@ -363,6 +363,7 @@ def _create_service(self):
 		self._dbusservice.add_path('/GensetProductId',
 			value=self._dbusmonitor.get_value(self._remoteservice, '/ProductId'))
 
+		self._dbusservice.register()
 		# We need to set the values after creating the paths to trigger the 'onValueChanged' event for the gui
 		# otherwise the gui will report the paths as invalid if we remove and recreate the paths without
 		# restarting the dbusservice.
